packages/teams-js/src/public/video.ts

```diff
@@ -216,4 +216,92 @@ export namespace video {
   export function isSupported(): boolean {
     return ensureInitialized(runtime) && runtime.supports.video ? true : false;
   }
+
+  /**
+   * Namespace to get video frames from a media stream
+   */
+  export namespace mediaStream {
+    export function isSupported(): boolean {
+      return ensureInitialized(runtime) && textureStreamAvailable() && !!runtime.supports.video?.mediaStream;
+    }
+
+    function textureStreamAvailable(): boolean {
+      return !!(window['chrome']?.webview?.getTextureStream && window['chrome']?.webview?.registerTextureStream);
+    }
+
+    export type ReceivedVideoFrame = {
+      videoFrame: VideoFrame;
+    };
+
+    /**
+     * Video effect change call back function definition.
+     * The video app should resolve the promise to notify a successfully processed video frame.
+     * The video app should reject the promise to notify a failure.
+     */
+    export type VideoFrameCallback = (receivedVideoFrame: ReceivedVideoFrame) => Promise<VideoFrame>;
+
+    type MediaStreamInfo = {
+      streamId: string;
+    };
+
+    export function registerForVideoFrame(frameCallback: VideoFrameCallback): void {
+      ensureInitialized(runtime, FrameContexts.sidePanel);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+
+      registerHandler('video.startVideoExtensibilityVideoStream', async (mediaStreamInfo: MediaStreamInfo) => {
+        // when a new streamId is ready:
+        const { streamId } = mediaStreamInfo;
+        const videoTrack = await getInputVideoTrack(streamId);
+        const generator = createProcessedStreamGenerator(videoTrack, frameCallback);
+        window['chrome']?.webview?.registerTextureStream(streamId, generator);
+      });
+    }
+
+    async function getInputVideoTrack(streamId: string): Promise<MediaStreamVideoTrack> {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const chrome = window['chrome'] as any;
+      const mediaStream = await chrome.webview.getTextureStream(streamId);
+
+      return mediaStream.getVideoTracks()[0];
```

Comment:
Is it possible that getVideoTracks return empty or undefined or null?

Reply:
If getTextureStream fails, it's better we send a message to TMP to persist telemetry and log.